packages/target-web3-v1/src/codegen/types.ts

```diff
@@ -1,26 +1,32 @@
+import { compact } from 'lodash'
 import { AbiOutputParameter, AbiParameter, EvmOutputType, EvmType, TupleType } from 'typechain'
 
-export function codegenInputTypes(input: AbiParameter[]): string {
+import { STRUCT_INPUT_POSTFIX, STRUCT_OUTPUT_POSTFIX } from '../common'
+
+interface GenerateTypeOptions {
+  returnResultObject?: boolean
+  useStructs?: boolean // uses struct type for first depth, if false then generates first depth tuple types
+}
+
+export function codegenInputTypes(options: GenerateTypeOptions, input: Array<AbiParameter>): string {
```

Comment:
Same feedback for codegenInputTypes (prefer `codegenInputTypes(input, options?)`)